UnityProject/Assets/Scripts/Tilemaps/Behaviours/TilemapDamage.cs

```diff
@@ -246,6 +246,31 @@ private void DoBulletDamage(BulletBehaviour bullet, Vector3 forceDir, Vector3 hi
 				}
 			}
 		}
+		if (bullet.isMiningBullet)
```

Comment:
Not a fan of this too. But adding new attack type will take a lot more effort because i will have to define in TilemapDamage damage to all sorts of armor/tiles.